pandas/src/joins_func_helper.pxi.in

```diff
@@ -158,3 +163,21 @@ def asof_join_{{on_dtype}}(ndarray[{{on_dtype}}] left_values,
 
 {{endfor}}
 
+
+#----------------------------------------------------------------------
+# stringify
+#----------------------------------------------------------------------
+
+def stringify(ndarray[object, ndim=2] xt):
+    cdef:
+        Py_ssize_t n
+        ndarray[object] result
+
+    n = len(xt)
+    result = np.empty(n, dtype=np.object)
+
+    for i in range(n):
+        result[i] = xt[i].tostring()
```

Comment:
I've added a couple comments to address this. When the `by` parameter has multiple entries, then we want to store the entire array in the hash table. Unfortunately, NumPy arrays aren't hashable. After lots of digging, the fastest thing to do is to convert to a string.